### DISCUSSION TITLE
Fish shell cursor setting doesn't work

### DISCUSSION THREAD

**Discussion by blt-r** [ghostty-org/ghostty#3887] [Feature Requests, Ideas] [answered]
I use vi mode in fish shell with different cursor shapes for different modes like this:
```fish
# config.fish
fish_vi_key_bindings
set fish_cursor_default     block      blink
set fish_cursor_insert      line       blink
set fish_cursor_replace_one underscore blink
set fish_cursor_visual      block
```

But in Ghostty the cursor is always a line in shell. If I do `shell-integration-features = no-cursor` or `shell-integration = none`, the cursor will always be a block. 

It works as expected in every other terminal I used.

**Comment by maurges** [comment]
I have a similar problem, but the cursor is always a line, and cannot be changed by terminal settings. Also the blinking randomly appears and disappears as I use some ncurses programs

**Comment by mahume** [comment]
Same issue here. Vi mode works in Kitty. Got Ghostty mostly setup with a lean config. Only issue I'm seeing so far is switching between shells (zsh and fish) has this same issue. Cursor seems to work in zsh, however not in fish. It just remains as a bar in both normal & insert mode

```
# conf.d/vi.fish
set fish_cursor_default     block      blink
set fish_cursor_insert      line       blink
set fish_cursor_replace_one underscore blink
set fish_cursor_visual      block
```

**Comment by 00-kat** [comment, marked as answer]
Plop this in your Fish config:
```sh
if status is-interactive
    if string match -q -- '*ghostty*' $TERM
        set -g fish_vi_force_cursor 1
    end
end
```
Once your copy of Fish has https://github.com/fish-shell/fish-shell/pull/10685, it'll work without this too.
